pkg/daemon/ceph/osd/encryption.go

```diff
@@ -172,6 +173,22 @@ func openEncryptedDevice(context *clusterd.Context, disk, target, passphrase str
 	return nil
 }
 
+func removeEncryptedDevice(context *clusterd.Context, target string) error {
+	args := []string{"remove", "--force", target}
+	output, err := context.Executor.ExecuteCommandWithTimeout(removeEncryptedDeviceCmdTimeOut, "dmsetup", args...)
+	// ignore error if no device was found.
+	if err != nil {
+		if !strings.Contains(output, "No such device") {
+			return errors.Wrapf(err, "failed to remove dm device %q: %q", target, output)
+		}
+		logger.Debugf("no stale dm device found at %q", target)
+	} else {
+		logger.Debugf("successfully removed stale dm device %q", target)
+	}
```

Comment:
@leseb does this sound fine ?

Reply:
Yes, thanks! I would just reverse the conditions and avoid the negative condition.
So use:

```go
if strings.Contains(output, "No such device") {
    logger.Debugf("no stale dm device found at %q", target)
}
return errors.Wrapf(err, "failed to remove dm device %q: %q", target, output)
```

Reply:
We don't need to return error if the device was not found right ? 

that would take the below form.
```go
	if err != nil {
		if !strings.Contains(output, "No such device") {
			logger.Debugf("no stale dm device found at %q", target)
			
			return nil
		}
		return errors.Wrapf(err, "failed to remove dm device %q: %q", target, output)
	}
	
	logger.Debugf("successfully removed stale dm device %q", target)
	return nil
```

Reply:
let me know the current one is fine or the above suggested form.